### PR TITLE
Clarify transform for ViewStyleProps, include children in union prop docs

### PR DESCRIPTION
**Pull request by brentvatne**
Show the proper shape of `transform` props:

screen shot 2015-05-06 at 6 05 12 pm

Currently it is just `transform: [object]`, so this should make it much clearer in the docs as well as properly validate.

cc @vjeux @ide

### CODE REVIEW

**Comment by ide** [comment]
Nice work there.

File: Libraries/Components/View/ViewStylePropTypes.js [hunk 1]
```diff
@@ -34,7 +34,15 @@ var ViewStylePropTypes = {
   ),
   shadowOpacity: ReactPropTypes.number,
   shadowRadius: ReactPropTypes.number,
-  transform: ReactPropTypes.arrayOf(ReactPropTypes.object),
+  transform: ReactPropTypes.arrayOf(
+    ReactPropTypes.oneOfType([
```
Comment (frantic) [inline Libraries/Components/View/ViewStylePropTypes.js]: Will this throw a warning if I use

```javascript
<View style={{transform: {rotation: 1, scaleX: 2}}} />
```
? (i.e. mixture of these shapes)

Comment (brentvatne) [inline Libraries/Components/View/ViewStylePropTypes.js]: @frantic - nope, but that won't work - you need to put each transform in a separate object. Also, I have realized that `rotation` is not a prop on iOS, it looks like that was just for Android. On iOS, the `rotate` transform property exists and accepts a string: `{rotate: "1deg"}` or `{rotate: "1rad"}` just like the CSS property.

So this is how that would be written:

```javascript
<View style={{transform: [{rotate: "1deg"}, {scaleX: 2}]}} />
```

If you were to writer this, rotate would be applied by scaleX would not:

```javascript
<View style={{transform: [{rotate: "1deg", scaleX: 2}]}} />
```

**Comment by brentvatne** [comment]
d177a45bdce124b635dbc31ab0949d2e9d139eb3 broke the transform docs, rebased and fixed so this is now in a new section:

screen shot 2015-05-07 at 12 18 24 pm
screen shot 2015-05-07 at 12 19 26 pm

**Pull request merged**